OSPSuite.Presentation.Importer/Services/IImporterTask.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Assets;
+using OSPSuite.Core.Importer;
+using OSPSuite.Presentation.Importer.Core.DataFormat;
+
+namespace OSPSuite.Presentation.Importer.Services
+{
+   public interface IImporterTask
+   {
+      int GetImageIndex(DataFormatParameter parameter);
+
+      string CheckWhetherAllDataColumnsAreMapped(IReadOnlyList<ColumnInfo> dataColumns, IEnumerable<DataFormatParameter> mappings);
+   }
+
+   public class ImporterTask : IImporterTask
+   {
+      public int GetImageIndex(DataFormatParameter parameter)
+      {
+         switch (parameter)
+         {
+            case MetaDataFormatParameter mp:
+               return ApplicationIcons.IconIndex(ApplicationIcons.MetaData);
+            case MappingDataFormatParameter mp:
+               return ApplicationIcons.IconIndex(ApplicationIcons.UnitInformation);
+            case GroupByDataFormatParameter gp:
+               return ApplicationIcons.IconIndex(ApplicationIcons.GroupBy);
+            default:
+               throw new Exception($"{parameter.GetType()} is not currently been handled");
+         }
+      }
+
+      public string CheckWhetherAllDataColumnsAreMapped(IReadOnlyList<ColumnInfo> dataColumns, IEnumerable<DataFormatParameter> mappings)
+      {
+         var subset = mappings.OfType<MappingDataFormatParameter>().ToList();
+         return (from col in dataColumns where col.IsMandatory where subset.FirstOrDefault(cm => cm.MappedColumn.Name.ToString() == col.Name) == null select col.Name).FirstOrDefault();
```

Comment:
why are you using this type of query suddenly? Feels like you are using LINQ everywhere else

Reply:
transformed to LINQ